sync/src/chain.rs

```diff
@@ -457,7 +457,10 @@ impl ChainSync {
 		if self.state != SyncState::WaitingPeers {
 			return;
 		}
-		let best_block = io.chain().chain_info().best_block_number;
+		// Make sure the snapshot block is not too far away from our best block, network best block and
+		// fork detection block
+		let mut best_block = max(io.chain().chain_info().best_block_number, self.highest_block.unwrap_or(0));
```

Comment:
isn't `highest_block` the highest block we've seen broadcasted? how will the check for `sn - best_block > SNAPSHOT_RESTORE_THRESHOLD` ever pass?